### ISSUE TITLE
[Feature]: Estimate Location from other photos

### ISSUE THREAD

**Issue by r3nor** [immich-app/immich#566]
### Feature detail

So Photoprism has a feature that given a 24h time span, it sets the geotag of the images from that span with the geolocation of other images that were taken the same day.

I think this is a very useful feature, but 24h I think is too much and can be innacurate. 

It would be a good feature if Immich could get Geolocation estimations from close-taken pictures for those that do not have Geolocation metadata in short timespans.

### Platform

Server

**Comment by alextran1502** [comment]
Hmm, since most of the picture uploaded from the phone already has geolocation in the image's EXIF. I am not sure if we need this info at all

**Comment by r3nor** [comment]
Yes but even with mobile not all photos may have geotags. I explain why I thought about this:

I went on vacation. I made some exterior photos to my family, but then we went to visit a cave where there was no GPS signal, the photos I took in there have no geotags. Also, later that same day we went to eat to a restaurant where there wasn't GPS signal too.

This is not a rare situation. In my case, I don't have any GPS signal on almost any interior spaces...

**Comment by EnderPicture** [comment]
To add, Google photos have this feature, though it also uses location history data. It would be helpful for phone + camera setups where a traveller would use phones to take snapshots while dedicated cameras with no GPS for more artistic/composed landscapes.
